llvm/tools/sycl-post-link/DeviceGlobals.cpp

```diff
@@ -0,0 +1,124 @@
+//===----- DeviceGlobals.cpp - SYCL Device Globals Pass -------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// See comments in the header.
+//===----------------------------------------------------------------------===//
+
+#include "DeviceGlobals.h"
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/Error.h"
+
+#include <cassert>
+
+using namespace llvm;
+
+namespace {
+
+constexpr StringRef SYCL_DEVICE_GLOBAL_SIZE_ATTR = "sycl-device-global-size";
+constexpr StringRef SYCL_UNIQUE_ID_ATTR = "sycl-unique-id";
+constexpr StringRef SYCL_DEVICE_IMAGE_SCOPE_ATTR = "device_image_scope";
+
+void AssertRelease(bool Cond, const char *Msg) {
+  if (!Cond)
+    report_fatal_error(Twine("DeviceGlobals.cpp: ") + Msg);
```

Comment:
Macro __FILE__ may be used for specifying source file name.

Reply:
Yes, I've changed the code to use the `__FILE__` macro.